smithy-cli/src/main/java/software/amazon/smithy/cli/commands/InitCommand.java

```diff
@@ -113,13 +171,45 @@ private void cloneTemplate(String repositoryUrl, String template, String directo
 
         exec(ListUtils.of("git", "checkout"), temp);
 
+        // Use templateName if directory is not specified
+        if (directory == null) {
+            directory = template;
+        }
+
         IoUtils.copyDir(Paths.get(temp.toString(), templatePath), Paths.get(directory));
 
         try (ColorBuffer buffer = ColorBuffer.of(env.colors(), env.stderr())) {
             buffer.println(String.format("Smithy project created in directory: %s", directory), ColorTheme.SUCCESS);
         }
     }
 
+    private static void loadSmithyTemplateJsonFile(String repositoryUrl, Path root, Path temp) {
```

Comment:
This works, but I think we could just checkout the whole repo in the future to the `build` directory, then do a `git update` each time. That would probably be much faster.